liblwgeom/measures3d.c

```diff
@@ -1133,63 +1164,46 @@ the plane is stored as a point in plane (plane.pop) and a normal vector (plane.p
 int
 define_plane(POINTARRAY *pa, PLANE3D *pl)
```

Comment:
I am not sure about the simplification of define_plane. The reason I did it this complicated was an attempt to avoid small numbers giving large errors. If the first 3 points is very close to each other a very small error (floating point error) can make the plane quite wrong which can give quite big error if the polygon is large. This is far away in my head no, since it was a long time ago. I will look more into it.

Reply:
Maybe we could try to add some test with really close numbers (but bigger than FP_TOLERANCE) to see if it reproduces it.

Reply:
Yes, I think we need to think about stability for cases like that and also "dirty" polygons, that has more or less coplanar vertex points.

Reply:
BTW, since now when we can't calculate the plane it defaults to checking the external ring itself as a line, we could add an extra check in `define_plane` and ignore any point whose distance to the previous ones is less than FP_TOLERANCE or something like this.